### PR TITLE
Add notes for 27050

### PR DESCRIPTION
**Pull request by dergoegge**
questions will follow later...

### CODE REVIEW

File: _posts/2023-03-22-#27050.md [hunk 1]
```diff
@@ -10,15 +10,59 @@ status: upcoming
...
+    dictate which chain a node follows. Meaning that a large reorg would be able
+    to orphan (parts of) the assumed-valid chain.
+
+- Nodes in prune mode (enabled by the `-prune` setting) fully download and
+  validate the chain history but only store a (configurable) portion of the
+  recent history.
+
+- [PR 27050](https://github.com/bitcoin/bitcoin/pull/27050) proposes to skip
+  downloading the witnesses for blocks prior to the configured assume-valid
+  point, for nodes running in prune mode. The rational for this change is that
```
Comment (glozow) [inline _posts/2023-03-22-#27050.md:57]:
```suggestion
  point, for nodes running in prune mode. The rationale for this change is that
```

File: _posts/2023-03-22-#27050.md [hunk 2]
```diff
@@ -10,15 +10,59 @@ status: upcoming
...
+  pruned nodes currently download witnesses but then (prior to the assume-valid
+  point) don't validate them and delete them shortly after. So why not skip
+  downloading those witnesses and save some bandwidth?
```
Comment (glozow) [inline _posts/2023-03-22-#27050.md:60]: maybe mention how much bandwidth because "some" kind of undersells it

File: _posts/2023-03-22-#27050.md [hunk 3]
```diff
@@ -10,15 +10,59 @@ status: upcoming
...
+
+  - Witnesses are commited to by placing the root of the witness merkle tree in
+    the block's coinbase transaction. By doing so, the witnesses are commited to
+    in the transaction merkle tree through the coinbase transaction. Nesting the
+    witness commitment in the coinbase transaction was done to make SegWit
+    soft-fork compatible.
+
+- `Assume-valid` is a node setting that makes the node skip transaction
+  validity checks (signature and script checks) prior to a pre-determined
+  "known to be good" block (assume-valid point). The default assume-valid point
```
Comment (glozow) [inline _posts/2023-03-22-#27050.md:31]: Maybe specify "known to be good" to be "has a lot of work"

File: _posts/2023-03-22-#27050.md [hunk 4]
```diff
@@ -10,15 +10,59 @@ status: upcoming
...
+    ([quoted](https://bitcoin.stackexchange.com/questions/59940/what-are-the-trust-assumptions-in-assumed-valid-in-bitcoin-core-0-14))
+    Additionally, as long as the full chain history remains available for
+    auditing it would be hard for such an attack to go unnoticed.
+
+  - It is also important to note that the configured assume-valid point does not
+    dictate which chain a node follows. Meaning that a large reorg would be able
+    to orphan (parts of) the assumed-valid chain.
+
+- Nodes in prune mode (enabled by the `-prune` setting) fully download and
+  validate the chain history but only store a (configurable) portion of the
```
Comment (glozow) [inline _posts/2023-03-22-#27050.md:52]:
```suggestion
  validate the chain history to build a UTXO set enabling them to fully validate any new transaction,
   but only store a (configurable) portion of the
```

File: _posts/2023-03-22-#27050.md [hunk 5]
```diff
@@ -10,15 +10,59 @@ status: upcoming
...
-
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
-## Questions
-1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)? What was your review approach?
--->
+- [BIP 141
+  (SegWit)](https://github.com/bitcoin/bips/blob/master/bip-0141.mediawiki)
+  introduced a new structure called a `witness` that is committed to blocks
```
Comment (glozow) [inline _posts/2023-03-22-#27050.md:17]: Slightly confusing if you don't mention that the witness is a new *transaction* field.
```suggestion
  introduced a new structure called a `witness` to each transaction. Transactions' witnesses are committed to blocks
```

File: _posts/2023-03-22-#27050.md (hunk 2 above)
Comment (dergoegge) [reply _posts/2023-03-22-#27050.md:60]: Was thinking of asking the participants how much bandwidth it saves in the questions. Is that to much homework?

File: _posts/2023-03-22-#27050.md (hunk 3 above)
Comment (dergoegge) [reply _posts/2023-03-22-#27050.md:31]: I don't think there is any work requirement on the assume-valid point itself, e.g. we could set it to a low-work point and that would still work (e.g. if we never update the value in a release).

**Comment by dergoegge** [comment]
Added some questions will add more later

File: _posts/2023-03-22-#27050.md [hunk 6]
```diff
@@ -10,15 +10,79 @@ status: upcoming
...
+   `0000000000000000000013a20dcc8577282e1eabd430592bb8afdd5fe544c05a`? (Hint:
+   the `getblock` RPC returns the `size` and `strippedsize` (size excluding
+   witnesses) for each block).
+
+3. The end goal of the PR can be achieved with very few changes to the code
+   (ignoring edge case scenarios). It essentially only requires two changes,
+   one to the block request logic and one to block validation. Can you (in your
+   own words) describe these two changes in more detail?
+
+4. While script validation is skipped under assume-valid, other checks that
```
Comment (glozow) [inline _posts/2023-03-22-#27050.md:79]:
```suggestion
4. Without this PR, while script validation is skipped under assume-valid, other checks that
```

**Comment by glozow** [comment]
Thanks @dergoegge! Notes look good, made some minor edits. Feel free to open another PR if you want to add more questions.

**Pull request merged**